week-8/Homework/mandatory/3-dog-photo-gallery/script.js

```diff
@@ -0,0 +1,25 @@
+// This function should retrieve the JSON from the `countryURL` and then call onCountryDataReceived() with the JSON
+function getData (imageURL) {
+  fetch (imageURL)
+    .then (response => {
+      if (response.status >= 200 && response.status <= 299) {
+        return response.json ();
+      } else {
+        throw new Error (
+          `Encountered something unexpected: ${response.status} ${response.statusText}`
+        );
+      }
+    })
+    .then (data => addImage (data))
+    .catch (error => console.log (error));
```

Comment:
This works really well for showing the developer the error information, but it doesn't really tell a normal user that anything has gone wrong. Can you think how to also let the user know what's happening / what to expect? Similarly for the humour question :)

Reply:
Thanks, Daniel I have updated all the recommendations now.

Reply:
I think this will still only appear in the console, right? Have a think about (thought up to you whether you want to implement) what kinds of things could you do which would show up to users :)

Reply:
Would you think that writting  if (!response.ok) {
          throw Error(response.statusText);
        } would be better?

Reply:
Where would you expect that to show up?

In general, the DOM is the mechanism we have to change the page from javascript so that users can see information we want to share

Reply:
So we can create p tags using Dom JS then we can display the error message there instead of the console log I would suggest this is the solution you are expecting isn't?

Reply:
Exactly :)